### DISCUSSION TITLE
How do scheduled pipeline of renovate-runner and schedule setting in renovate.json work together?

### DISCUSSION THREAD

**Discussion by andorfermichael** [renovatebot/renovate#10533] [Request Help] [answered]
I am currently using the [renovate-runner](https://gitlab.com/renovate-bot/renovate-runner) with a scheduled pipeline (as described in the repository [readme](https://gitlab.com/renovate-bot/renovate-runner)).

I have seen that it is also possible to define [a schedule](https://docs.renovatebot.com/configuration-options/#schedule) in the renovate.json.

Do they work together, and if yes, how?

Does the pipeline only trigger the check of the renovate.json? If so does the pipeline schedule another pipeline run automatically?

**Comment by rarkins** [comment, marked as answer]
The bot runs as often as your pipeline schedule says. Then, while it's processing a repo it checks if the current timestamp falls within the allowed repo schedule.

So for example if scheduled your pipeline to run once each Saturday, but then configured schedules in your repos for "before 3am on Monday" then nothing would ever get done as there's no overlap.

**Reply by andorfermichael** [reply]
I see. Thanks for clarifying.

**Reply by viceice** [reply]
Best practise is to run the bot on hourly schedule and use renovate schedules to limit updates. So renovate can update automerge prs when required.
